src/main/java/gregtech/api/gui/widgets/CycleButtonWidget.java

```diff
@@ -132,7 +132,14 @@ public void readUpdateInfo(int id, PacketBuffer buffer) {
     public boolean mouseClicked(int mouseX, int mouseY, int button) {
         super.mouseClicked(mouseX, mouseY, button);
         if (isMouseOverElement(mouseX, mouseY)) {
-            this.currentOption = (currentOption + 1) % optionNames.length;
+            //Allow only the RMB to reverse cycle
+            if(button == 1) {
+                //Wrap from the first option to the last if needed
+                this.currentOption = currentOption - 1 < 0 ? currentOption + optionNames.length - 1 : currentOption - 1;
```

Comment:
This seems bit overcomplicated to me, so there is simplification
```suggestion
                this.currentOption = currentOption == 0 ? optionNames.length - 1 : currentOption - 1;
```

Reply:
I have applied this change.